Sources/Orders/OrdersServiceCustom.swift

```diff
@@ -111,27 +82,33 @@ public final class OrdersServiceCustom<O, D, R: OrdersRegistrationModel, E: Erro
             isDefault: false
         )
 
-        let pushAuth = v1.grouped(middleware)
+        let v1 = app.grouped("api", "orders", "v1")
+        v1.get("devices", ":deviceIdentifier", "registrations", ":orderTypeIdentifier", use: { try await self.ordersForDevice(req: $0) })
+        v1.post("log", use: { try await self.logError(req: $0) })
+
+        let v1auth = v1.grouped(AppleOrderMiddleware<O>())
+        v1auth.post("devices", ":deviceIdentifier", "registrations", ":orderTypeIdentifier", ":orderIdentifier", use: { try await self.registerDevice(req: $0) })
+        v1auth.get("orders", ":orderTypeIdentifier", ":orderIdentifier", use: { try await self.latestVersionOfOrder(req: $0) })
+        v1auth.delete("devices", ":deviceIdentifier", "registrations", ":orderTypeIdentifier", ":orderIdentifier", use: { try await self.unregisterDevice(req: $0) })
         
-        pushAuth.post("push", ":orderTypeIdentifier", ":orderIdentifier", use: { try await self.pushUpdatesForOrder(req: $0) })
-        pushAuth.get("push", ":orderTypeIdentifier", ":orderIdentifier", use: { try await self.tokensForOrderUpdate(req: $0) })
+        if let pushRoutesMiddleware {
+            let pushAuth = v1.grouped(pushRoutesMiddleware)
+            pushAuth.post("push", ":orderTypeIdentifier", ":orderIdentifier", use: { try await self.pushUpdatesForOrder(req: $0) })
+            pushAuth.get("push", ":orderTypeIdentifier", ":orderIdentifier", use: { try await self.tokensForOrderUpdate(req: $0) })
+        }
     }
 }
 
 // MARK: - API Routes
 extension OrdersServiceCustom {
     func latestVersionOfOrder(req: Request) async throws -> Response {
         logger?.debug("Called latestVersionOfOrder")
-
-        guard FileManager.default.fileExists(atPath: delegate.zipBinary.unixPath()) else {
-            throw Abort(.internalServerError, suggestedFixes: ["Provide full path to zip command"])
-        }
         
-        let dateFormatter = ISO8601DateFormatter()
-        dateFormatter.formatOptions = .withInternetDateTime
-
+        let dateFormatter = DateFormatter()
+        dateFormatter.locale = Locale(identifier: "en_US_POSIX")
+        dateFormatter.timeZone = TimeZone(secondsFromGMT: 0)
+        dateFormatter.dateFormat = "EEE, dd MMM yyyy HH:mm:ss zzz"
```

Comment:
These are not cheap to create - it might be worth caching this inside Vapor's storage

Reply:
You mean like [this](https://docs.vapor.codes/advanced/services/?h=#writable)?

Reply:
Yeah

Reply:
I have added it [here](https://github.com/vapor-community/PassKit/pull/10/commits/a52e732943085087a671c2f71d3cf323f89e955d#diff-14ab372d263b20a84907c52d47cd29e5cb3191692a7db06c02fdc41317a7120a).
There is the following `Sendable` warning on `ISO8601Key`, how do I fix it?
`Conformance of 'ISO8601DateFormatter' to 'Sendable' is unavailable`